### PR TITLE
update site docs

### PR DESCRIPTION
**Pull request by tinashechiraya**

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 42 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 31 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 32 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 41 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 40 sec

File: docs/src/developer/observations.md [hunk 1]
```diff
@@ -18,24 +18,45 @@ The Observations API facilitates users in creating observations for various site
 
 The API expects a JSON payload containing the following fields:
 
-- `datainput`: Object containing observation data.
+- `create_site_or_observation`: variable containing true or false value on whether to create a new site and attach an observation to it or use an existing one.
+- `siteId`: interger value provided create_site_or_observation is false.
+- `image_0`: these can be added from _0 to _10 or more. This is for attaching site images which will be retrived from the request object.
```
Comment (Jeremy-Prior) [inline docs/src/developer/observations.md]:
```suggestion
- `image_0`: these can be added from _0 to _10 or more. This is for attaching site images which will be retrieved from the request object.
```

File: docs/src/developer/observations.md [hunk 2]
```diff
@@ -18,24 +18,45 @@ The Observations API facilitates users in creating observations for various site
 
 The API expects a JSON payload containing the following fields:
 
-- `datainput`: Object containing observation data.
+- `create_site_or_observation`: variable containing true or false value on whether to create a new site and attach an observation to it or use an existing one.
+- `siteId`: interger value provided create_site_or_observation is false.
```
Comment (Jeremy-Prior) [inline docs/src/developer/observations.md]:
```suggestion
- `siteId`: integer value provided create_site_or_observation is false.
```

File: docs/src/developer/sites.md [hunk 3]
```diff
@@ -111,10 +114,36 @@ Any field can be updated on the site
...
+        'images': image_files,
+        'user_id': 1
+    }
+    ```
+
+#### Responses
+  - HTTP_201_CREATED: when the image/s is/are successfully saved.
+  - HTTP_400_BAD_REQUEST: invalid site id supplied (non integer ,unconvertable string char etc).
+  - HTTP_404_NOT_FOUND: no site exist with specified id.
+  - HTTP_500_INTERNAL_SERVER_ERROR: provided image is invalid or something unexpected occured.
```
Comment (Jeremy-Prior) [inline docs/src/developer/sites.md]:
```suggestion
  - HTTP_500_INTERNAL_SERVER_ERROR: provided image is invalid or something unexpected occurred.
```

File: docs/src/developer/sites.md [hunk 4]
```diff
@@ -111,10 +114,36 @@ Any field can be updated on the site
...
+    ```
+    {
+        'images': image_files,
+        'user_id': 1
+    }
+    ```
+
+#### Responses
+  - HTTP_201_CREATED: when the image/s is/are successfully saved.
+  - HTTP_400_BAD_REQUEST: invalid site id supplied (non integer ,unconvertable string char etc).
```
Comment (Jeremy-Prior) [inline docs/src/developer/sites.md]:
```suggestion
  - HTTP_400_BAD_REQUEST: invalid site id supplied (non integer, unconvertable string char etc).
```

File: docs/src/developer/observations.md [hunk 5]
```diff
@@ -268,55 +306,87 @@ This API endpoint allows retrieving, updating, or deleting a specific observatio
...
 
 ### Additional Information
 
-1. **image_0:** Represents a site image. If uploading multiple site images, it would be `image_0`, `image_1`, etc.
-2. For attaching images to pests/groups (e.g. flatworms, stoneflies, other_mayflies), the variable should be named `"pest_0:name_of_pest"`. The value should be the binary file (image), and the corresponding pest/group must be set to `true`.
-3. Accommodations for file uploading should be made on the frontend. Simply call the API for observation creation, passing the required parameters, and all the data will be saved on that site.
+1. **pest_0:10** Represents a pest image. If uploading multiple pest images, it would be `pest_0`, `pest_1`, etc.
+-  ":10" is the id of the group to which the pest should be uploaded to.
+- ``` grouping ids 
+    bugs or beetles: 10 (staging) , 23 (production)
```
Comment (Jeremy-Prior) [inline docs/src/developer/observations.md]:
```suggestion
    bugs or beetles: 10 (staging), 23 (production)
```

File: docs/src/developer/observations.md [hunk 6]
```diff
@@ -268,55 +306,87 @@ This API endpoint allows retrieving, updating, or deleting a specific observatio
...
+
+#### Endpoint: `/monitor/observations/<int:observationId>/save-images/`
+
+#### Request Type: Post Method
+
+#### accepted formats: MultiPart, Form, JSON
+
+- Example payload
+```json
+"observationdId": 1 , //the id of the observation to which the images will be attached
```
Comment (Jeremy-Prior) [inline docs/src/developer/observations.md]:
```suggestion
"observationdId": 1, //the id of the observation to which the images will be attached
```

**Review by Jeremy-Prior — CHANGES_REQUESTED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 40 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 51 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 40 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 27 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 39 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 41 sec

**Comment by tinashechiraya** [comment]
@Jeremy-Prior I have committed you suggestions. Thanks

**Review by Jeremy-Prior — APPROVED** [review]
LGTM

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 1 min 36 sec

**Pull request merged**